src/sign/signature.rs

```diff
@@ -772,7 +787,7 @@ mod test {
 
         aggregator.include_signer(1, p1_public_comshares.commitments[0], &p1_sk.to_public());
 
-        let signers = aggregator.get_signers();
+        aggregator.dedup_signers().unwrap();
```

Comment:
I actually prefer to have the deduplication being done while getting the internal list of signers, as this could easily be overlooked say if the library was to be refactored, or rendered more modular for external parties to accommodate differently the signing process. This could lead to situations where deduplication is forgotten which would be bad.

Maybe just changing the name to something explicit enough about the internal behavior is just what's needed here?

Reply:
How about this: we lock down the API so that signers can only ever be added by an api call (so no `push` or setting the member with a `Vec`), then we can guarantee the deduplication on insert and no longer need to have nasty side effects in getters?

Reply:
Maybe it's as easy as mandating the use of `include_signer`?

Reply:
Yeah I like that !

Reply:
Done.